src/dotnet/ExecutablePackageObtainer/ToolConfiguration.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.DotNet.ExecutablePackageObtainer
+{
+    public class ToolConfiguration
+    {
+        public ToolConfiguration(
+            string commandName,
+            string toolAssemblyEntryPoint)
+        {
+            if (string.IsNullOrWhiteSpace(commandName))
+            {
+                throw new ArgumentNullException(nameof(commandName), "Cannot be null or whitespace");
+            }
+
+            EnsureNoInvalidCharacters(commandName, nameof(toolAssemblyEntryPoint));
+
+            if (string.IsNullOrWhiteSpace(toolAssemblyEntryPoint))
+            {
+                throw new ArgumentNullException(nameof(toolAssemblyEntryPoint), "Cannot be null or whitespace");
+            }
+
+            CommandName = commandName;
+            ToolAssemblyEntryPoint = toolAssemblyEntryPoint;
+        }
+
+        private void EnsureNoInvalidCharacters(string commandName, string nameOfParam)
```

Comment:
Should this be called `EnsureNoInvalidFilenameCharacters` to make the intended list of invalid characters more apparent?

Reply:
Fixed